UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs

```diff
@@ -122,11 +131,53 @@ public void ApplyTraumaDamage(float tramuaDamage, TraumaticDamageTypes damageTyp
 				if (damageType == TraumaticDamageTypes.PIERCE) { currentPierceDamage += MultiplyTraumaDamage(tramuaDamage); }
 				CheckCutSize();
 			}
-			//Burn damage checks for it's own armor damage type.
+			//Burn and blunt damage checks for it's own armor damage type.
 			if (damageType == TraumaticDamageTypes.BURN)
 			{
 				TakeBurnDamage(MultiplyTraumaDamage(tramuaDamage));
 			}
+
+			if (damageType == TraumaticDamageTypes.BLUNT)
+			{
+				if (DMMath.Prob(SelfArmor.Melee * 100) == false)
+				{
+					TakeBluntDamage(tramuaDamage);
+				}
+			}
+		}
+
+		public void TakeBluntDamage(float damage)
+		{
+			foreach (ItemSlot slot in OrganStorage.GetIndexedSlots())
+			{
+				if (!slot.IsEmpty)
+				{
+					if (slot.Item.gameObject.TryGetComponent<BodyPart>(out var bodyPart))
+					{
+						if (bodyPart.CanBeBroken)
+						{
+							bodyPart.health -= damage;
+							bodyPart.CheckIfBroken(true);
+						}
+					}
+				}
+			}
+		}
+
+		public void CheckIfBroken(bool announceHurtDamage = false)
+		{
+			if (CanBeBroken)
+			{
+				if (Severity == BoneFracturesOnDamageSevarity) { isFractured = true; }
+				if (Severity >= BoneBreaksOnDamageSevarity) { isBroken = true; }
+
+				if (isFractured && isBroken != true && announceHurtDamage)
+				{
+					Chat.AddActionMsgToChat(HealthMaster.gameObject,
```

Comment:
would be cool to also play the sound

Reply:
not really a request, more like a thought. You can resolve this one if you don't feel like adding the sound.

Reply:
I can't find a sound to use from /tg/ sadly so I'm closing this, but will keep it on my long list of to-do list for the future.